pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+)
+
+type tlsValidationResult struct {
+	Valid   bool
+	Message string
```

Comment:
I don't see the message being used ever

Reply:
this makes me think we only need a return type of bool, error for validateTLSConfig

Reply:
`Message` is used on line 162 where it's logged for the user to see.

Reply:
@fabianlindfors why don't we log it when the error is happening in validateTLSConfig, and have a simple bool, error return type

Reply:
@LouisPlisso We definitely could but I mostly put it on that level to stay consistent with the other error logs, they all appear in the reconciliation loop. You think we should break that pattern for now? It will be fixed once we find a better way to handle the reconciliation result with CLOUDP-66375.